src/commands/whoami/mod.rs

```diff
@@ -1,20 +1,56 @@
+use crate::http;
 use crate::settings::global_user::GlobalUser;
-use crate::terminal::{emoji, message};
+use crate::terminal::emoji;
+
+use cloudflare::endpoints::account;
+use cloudflare::framework::apiclient::ApiClient;
+use cloudflare::framework::HttpApiClientConfig;
+
+use prettytable::{Cell, Row, Table};
 
 pub fn whoami(user: &GlobalUser) -> Result<(), failure::Error> {
     // If using email + API key for auth, simply prints out email from config file.
-    let email: String = match user {
-        GlobalUser::GlobalKeyAuth { email, .. } => email.to_string(),
-        GlobalUser::TokenAuth { .. } => failure::bail!(
-            "At the moment, Wrangler cannot get user information for users using API tokens"
-        ),
+    let table: String = match user {
+        GlobalUser::GlobalKeyAuth { email, .. } => {
+            println!(
+                "\n{} You are logged in with a Global API Key, associated with the email '{}'.\n",
+                emoji::WAVING,
+                email,
+            );
+
+            fetch_accounts(user)?
+        }
+        GlobalUser::TokenAuth { .. } => {
+            println!("\n{} You are logged in with an API Token.\n", emoji::WAVING,);
+
+            fetch_accounts(user)?
+        }
     };
 
-    let msg = format!(
-        "{} You are logged in with the email '{}'.",
-        emoji::WAVING,
-        email
-    );
-    message::info(&msg);
+    println!("{}", &table);
     Ok(())
 }
+
+fn fetch_accounts(user: &GlobalUser) -> Result<String, failure::Error> {
+    let client = http::cf_v4_api_client(user, HttpApiClientConfig::default())?;
+    let response = client.request(&account::ListAccounts { params: None })?;
+
+    let mut table = Table::new();
+    let table_head = Row::new(vec![Cell::new("Account Name"), Cell::new("Account ID")]);
+    table.add_row(table_head);
+
+    match user {
+        GlobalUser::TokenAuth { .. } => {
+            if response.result.len() == 0 {
+                println!("Your token is missing the Account Settings: Read permission.\n\nPlease generate and auth with a new token that has these perms to be able to list your accounts.\n");
```

Comment:
is this really the only way you learn this? interesting.
are there any other error cases we need to account for in the response? and is it worth including at least info log lines for those errors?

Reply:
as far as i can tell yes

Reply:
i dont know what you mean by other errors

Reply:
like any non-200 response from the api. we have format_errors that will take a function to match against the code, but you could just pass it a function that prints the error itself.

https://github.com/cloudflare/wrangler/blob/master/src/http.rs#L91